### PR TITLE
feat(params): allow [_-] in url params

### PR DESCRIPTION
**Pull request by CPatchane**
## Status

**READY**

## Description
Allow to use `_` and `-` in path params name.

```yaml
# openapi.yml
paths:
  /api/demo/{id}/index
    ...
  /api/demo/users/{user_id}/index
    ...
```

Before:
```ts
export const getApiDemoIdIndex = (id: string) => {
  return client<GetApiDemoIdIndex200>({
    url: `/api/demo/${id}/index`,
    method: "get",
  })
}

// Won't work at all, but the function naming is also breaking:
export const getApiDemoUseridIndex = ...
```

After:
```ts
export const getApiDemoIdIndex = (id: string) => {
  return client<GetApiDemoIdIndex200>({
    url: `/api/demo/${id}/index`,
    method: "get",
  })
}

export const getApiDemoUsersUserIdIndex = (userId: string) => {
  return client<GetApiDemoUsersUserIdIndex200>({
    url: `/api/demo/users/${userId}/index`,
    method: "get",
  })
}
```

### CODE REVIEW

File: src/core/getters/route.ts [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
-  return path.split('').reduce((acc, letter) => {
-    if (letter === '{') {
-      return acc + '${';
-    }
-
-    if (letter === '}') {
-      return acc + '}';
-    }
-
-    return acc + sanitize(letter, { dot: true });
```
Comment (CPatchane) [inline src/core/getters/route.ts:13]: We were allowing dot here but it can't work since in `params.ts` we wasn't allowing it 🤷

Comment (anymaniax) [reply src/core/getters/route.ts:13]: dot are not for the params

Comment (anymaniax) [reply src/core/getters/route.ts:13]: you don't handle the case where you have something like this /{petId}.js

File: src/utils/string.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export const sanitize = (
   },
 ) => {
   const { whitespace = '', underscore = '', dot = '', dash = '' } = options || {};
-  let newValue = value.replace(/[^\w\s.-]/g, '');
```
Comment (anymaniax) [inline src/utils/string.ts]: don't need to add the _ was already working like that normally

Comment (CPatchane) [reply src/utils/string.ts]: Indeed, TIL `underscore` is included in `\w`, my bad

File: src/core/getters/route.ts (hunk 1 above)
Comment (CPatchane) [reply src/core/getters/route.ts:13]: Indeed, I rewrote that part and I added jest to be able to have unit tests for this, much simpler to test and make it robust. Wdyt?

File: jest.config.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (anymaniax) [inline jest.config.js]: really great idea. Would you be ok to use https://vitest.dev/ instead?

Comment (CPatchane) [reply jest.config.js]: Didn't know vitest but let's do it 👍

Comment (CPatchane) [reply jest.config.js]: Much simpler to config, it's done :)

**Comment by anymaniax** [comment]
Would be great to also run the test on pre-commit and in the ci

**Comment by CPatchane** [comment]
> Would be great to also run the test on pre-commit and in the ci

Indeed, done :)

**Review by anymaniax — APPROVED** [review]
Really nice job! 🎉

**Pull request merged**